contracts/RelayHub.sol

```diff
@@ -398,40 +442,40 @@ contract RelayHub is IRelayHub {
 
         // First preRelayedCall is executed.
         // Note: we open a new block to avoid growing the stack too much.
-        bytes memory data = abi.encodeWithSelector(
+        atomicData.data = abi.encodeWithSelector(
             IGasSponsor(address(0)).preRelayedCall.selector, recipientContext
         );
-
-        // preRelayedCall may revert, but the recipient will still be charged: it should ensure in
-        // acceptRelayedCall that this will not happen.
-        (bool success, bytes memory retData) = relayRequest.relayData.gasSponsor.call.gas(PRE_RELAYED_CALL_MAX_GAS)(data);
-
-        if (!success) {
-            revertWithStatus(RelayCallStatus.PreRelayedFailed);
+        bytes memory retData;
```

Comment:
Is there a reason to define retData outside and success inside the new block? it seems that you could move the line `atomicData.preReturnValue = abi.decode(retData, (bytes32));` inside the block as well, and you don't use retData elsewhere.

Reply:
No, no reason whatsoever. Not a bug though, so I guess does not require fixing, what do you think?